### PR TITLE
Fixes #12274

### LINKED ISSUE #12274: Switchtools don't check if they have a tool out already
If you spam attack_self on a switchtool, you can have a bunch of tools out at once:

gaycunt

It only works as the most recently activated tool, though.

### PR DESCRIPTION
**Pull request by ghost**
Fixes #12274

### CODE REVIEW

File: code/game/objects/items/weapons/switchtool.dm [hunk 1]
```diff
@@ -126,12 +126,15 @@
 
 	if(!stored_modules[module])
 		return
+	if(deployed)
+		return
```
Comment (PJB3005) [inline code/game/objects/items/weapons/switchtool.dm]: Make this return `FALSE` excplicitly.

**Review by PJB3005 — CHANGES_REQUESTED** [review]

File: code/game/objects/items/weapons/switchtool.dm (hunk 1 above)
Comment (ghost) [reply code/game/objects/items/weapons/switchtool.dm]: Why though

Comment (PJB3005) [reply code/game/objects/items/weapons/switchtool.dm]: Because this function should return a boolean, not either `TRUE` or `null`.

Comment (ghost) [reply code/game/objects/items/weapons/switchtool.dm]: That's a non standard formatting change

Comment (PJB3005) [reply code/game/objects/items/weapons/switchtool.dm]: Not really.

Comment (ghost) [reply code/game/objects/items/weapons/switchtool.dm]: Yes really

Comment (PJB3005) [reply code/game/objects/items/weapons/switchtool.dm]: No really.

Comment (ghost) [reply code/game/objects/items/weapons/switchtool.dm]: Yes really

Comment (Exxion) [reply code/game/objects/items/weapons/switchtool.dm]: It's not even a formatting change at all

**Comment by Probe1** [comment]
Re open when you address the review.

File: code/game/objects/items/weapons/switchtool.dm (hunk 1 above)
Comment (9600bauds) [reply code/game/objects/items/weapons/switchtool.dm]: OK, I'll bite
Literally what difference would this ever make

Comment (ComicIronic) [reply code/game/objects/items/weapons/switchtool.dm]: 1. This function is returning boolean values, so it should use that.
2. This function was converted from not returning anything, so it is not obvious in the code unless you read all the return statements that something actually comes out.
3. We hate Celt.

Comment (Exxion) [reply code/game/objects/items/weapons/switchtool.dm]: null and 0 are also not equal, and if you're a complete idiot/mad genius, you can exploit that

**Comment by gbasood** [comment]
this whole file needs to be reformatted eventually if we want to keep that return convention btw
it's all shit

**Comment by ComicIronic** [comment]
Harsh

**Pull request merged**